src/catch2/internal/catch_textflow.cpp

```diff
@@ -26,117 +26,225 @@ namespace {
         return std::memchr( chars, c, sizeof( chars ) - 1 ) != nullptr;
     }
 
-    bool isBoundary( std::string const& line, size_t at ) {
-        assert( at > 0 );
-        assert( at <= line.size() );
-
-        return at == line.size() ||
-               ( isWhitespace( line[at] ) && !isWhitespace( line[at - 1] ) ) ||
-               isBreakableBefore( line[at] ) ||
-               isBreakableAfter( line[at - 1] );
-    }
-
 } // namespace
 
 namespace Catch {
     namespace TextFlow {
+        void AnsiSkippingString::preprocessString() {
+            for ( auto it = m_string.begin(); it != m_string.end(); ) {
+                // try to read through an ansi sequence
+                while ( it != m_string.end() && *it == '\033' &&
+                        it + 1 != m_string.end() && *( it + 1 ) == '[' ) {
+                    auto cursor = it + 2;
+                    while ( cursor != m_string.end() &&
+                            ( isdigit( *cursor ) || *cursor == ';' ) ) {
+                        ++cursor;
+                    }
+                    if ( cursor != m_string.end() && *cursor == 'm' ) {
+                        // 'm' -> 0xff
+                        *cursor = AnsiSkippingString::sentinel;
+                        // if we've read an ansi sequence, set the iterator and
+                        // return to the top of the loop
+                        it = cursor + 1;
+                    } else {
+                        break;
```

Comment:
I think this would be more readable switched around, like so

```cpp
if ( cursor == m_string.end() || ... ) { break; }
*cursor = sentinel;
it = cursor + 1;
```